datafusion/sqllogictest/test_files/order.slt

```diff
@@ -985,13 +985,20 @@ drop table ambiguity_test;
 statement ok
 create table t(a0 int, a int, b int, c int) as values (1, 2, 3, 4), (5, 6, 7, 8);
 
-# expect this query to run successfully, not error
-query III
+# b is not selected in subquery
+query error DataFusion error: Schema error: No field named b\. Valid fields are t1\.c, t1\.a, t1\.a0\.
 select * from (select c, a, NULL::int as a0 from t order by a, c) t1
 union all
 select * from (select c, NULL::int as a, a0 from t order by a0, c) t2
 order by c, a, a0, b
```

Comment:
This query is executable in `main`, but it is not because `b` is not in the subquery

Reply:
I agree with you. This SQL isn't valid. However, it's added by #12721, which shows that issue #12446 has been solved. I do not understand the details of #12446. Maybe @alamb knows why we expect it works?

Reply:
I will check and get back to you

Reply:
I agree the query is invalid (and postgres agrees). I made a PR to fix it here:
- https://github.com/apache/datafusion/pull/15131